lib/matplotlib/backends/backend_pdf.py

```diff
@@ -1146,6 +1148,23 @@ def embedTTFType42(font, characters, descriptor):
             wObject = self.reserveObject('Type 0 widths')
             toUnicodeMapObject = self.reserveObject('ToUnicode map')
 
+            print(f"SUBSET {filename} characters: "
+                  f"{''.join(chr(c) for c in characters)}")
+            fontdata = self.getSubset(
+                filename,
+                ''.join(chr(c) for c in characters)
+            )
+            print(f'SUBSET {filename} {os.stat(filename).st_size}'
+                  f' ↦ {len(fontdata)}')
+
+            # reload the font object from the subset
+            # (all the necessary data could probably be obtained directly
+            # using fontLib.ttLib)
+            with tempfile.NamedTemporaryFile(suffix='.ttf') as tmp:
+                tmp.write(fontdata)
+                tmp.seek(0, 0)
+                font = FT2Font(tmp.name)
```

Comment:
Reloading the FT2Font object is a bit ugly, and I think it is only needed here to get the glyph widths, the cid to gid map and the unicode mapping. These could probably be obtained otherwise. On the other hand, reusing the old code makes this patch smaller.